engine/src/sync.rs

```diff
@@ -469,11 +505,7 @@ impl TransactionBatchOutcome {
                     .iter()
                     .chain(last_outcome.iter().map(|x| x.as_ref()));
                 for tx_outcome in all_outcomes {
-                    storage.revert_transaction_included(
-                        tx_outcome.hash,
-                        &tx_outcome.info,
-                        &tx_outcome.diff,
-                    )?
+                    tx_outcome.commit(storage)?
```

Comment:
Would it be better to add in the future a method to the `Storage` that will accept an array of txs and persist the changes atomically via `WriteBatch`?